design/restic-backup-and-restore-progress.md

```diff
@@ -0,0 +1,114 @@
+# Progress reporting for restic backups and restores
+
+Status: Draft
+
+During long-running restic backups/restores, there is no visibility into what (if anything) is happening, making it hard to know if the backup/restore is making progress or hung, how long the operation might take, etc.
+We should capture progress during restic operations and make it user-visible so that it's easier to reason about.
+This document proposes an approach for capturing progress of backup and restore operations and exposing this information to users.
+
+## Goals
+
+- Provide basic visibility into restic operations to inform users about their progress.
+
+## Non Goals
+
+- Capturing progress for non-restic backups and restores.
+
+## Background
+
+(Omitted, see introduction)
+
+## High-Level Design
+
+### restic backup progress
+
+The `restic backup` command provides progress reporting to stdout in JSON format, which includes the completion percentage of the backup.
+This progress will be read on some interval and the PodVolumeBackup Custom Resource's (CR) status will be updated with this information.
+
+### restic restore progress
+
+The `restic ls` command returns the number of files in a backup, including the size of each file.
```

Comment:
let's update this to describe the `restic stats` approach.

Reply:
Oops, missed this

Reply:
updated!